### PR TITLE
[MM-48161] Change data types for collections and topics

### PR DESCRIPTION
**Pull request by iomodo**
#### Summary
PR changes data types to `collectionTypes` and `topicTypes` maps to map[string]string like
collectionType -> pluginId
topicType -> collectionType
This makes them automatically unique and easier to access later.

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-48161

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by neallred** [comment]
@iomodo I'm not familiar with this code or the epic this is tied to. Whoever else is working on threads everywhere is probably a better fit.

File: app/collection.go [hunk 1]
```diff
@@ -17,29 +16,20 @@ func (a *App) registerCollectionAndTopic(pluginID, collectionType, topicType str
 	defer a.ch.collectionAndTopicTypesMut.Unlock()
 
 	// check if collectionType was already registered by other plugin
-	for existingPluginID, existingCollectionTypes := range a.ch.collectionTypes {
-		if existingPluginID != pluginID && utils.StringInSlice(collectionType, existingCollectionTypes) {
-			return model.NewAppError("registerCollectionAndTopic", "app.collection.add_collection.exists.app_error", nil, "", http.StatusBadRequest)
-		}
+	existingPluginID, ok := a.ch.collectionTypes[collectionType]
+	if existingPluginID != pluginID && ok {
```
Comment (lieut-data) [inline app/collection.go:20]: For clarity, perhaps reversing the check?
```suggestion
	existingPluginID, ok := a.ch.collectionTypes[collectionType]
	if ok && existingPluginID != pluginID {
```

**Review by lieut-data — APPROVED** [review]
Nice!

**Review by calebroseland — APPROVED** [review]

**Pull request merged**